django_informixdb/base.py

```diff
@@ -253,6 +254,39 @@ def get_new_connection(self, conn_params):
 
         return self.connection
 
+    def _get_connection_with_retries(self, connection_string, conn_params):
+        """
+        Attempt to open a connection, retrying on failure with an exponential backoff.
+        """
+        retry_params = conn_params.get("CONNECTION_RETRY", {})
+        max_attempts = retry_params.get("MAX_ATTEMPTS", 1)
+        wait_min = retry_params.get("WAIT_MIN", 0)
+        wait_max = retry_params.get("WAIT_MAX", 1000)
+        multiplier = retry_params.get("WAIT_MULTIPLIER", 25)
+        exp_base = retry_params.get("WAIT_EXP_BASE", 2)
+
+        attempt = 0
+        while True:
+            attempt += 1
+            try:
+                conn = pyodbc.connect(connection_string, autocommit=conn_params["AUTOCOMMIT"])
+            except pyodbc.Error as err:
```

Comment:
This seems quite broad - will this result in retries if eg. the connection string is misconfigured?

Reply:
Good point. I've updated the PR to only retry on error codes `-908` and `-27001`.